src/rust/engine/src/tasks.rs

```diff
@@ -45,36 +45,46 @@ impl rule_graph::Rule for Rule {
       &Rule::Intrinsic(_) => false,
     }
   }
+
+  fn is_intrinsic(&self) -> bool {
+    match self {
+      &Rule::Task(_) => false,
+      &Rule::Intrinsic(_) => true,
+    }
+  }
 }
 
 impl fmt::Display for Rule {
   fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
```

Comment:
As explained in the overall review comment, I do not think we want to change any of this function because that means changing the rule graph errors too. Instead, I recommend a new method like `fmt_for_visualization`.

Reply:
I ended up keeping the rule graph display the same as per our previous discussion, but I've kept modifications to this function, because I extracted it out into the shared `extract_task_elements()` helper method.